### PR TITLE
Multichain integration tests

### PR DESCRIPTION
**Pull request by estebgonza**
Integration tests can take configurations by networks.
(I tried to simplify the configuration code compared to my previous PR, it could be made clearer.)

- I've set up all our L2s except for Base as it's not supported.
- Fix on the L2 fees multicall, the decoder wasn't correct.

All tests pass, I make sure to test each farming pair by adding tests with native tokens.

### CODE REVIEW

**Comment by estebgonza** [comment]
```bash
PASS  src/dex/smardex/smardex-integration.test.ts (20.502 s)
  Smardex Integration Tests on ethereum
    ✓ getTopPoolsForToken (559 ms)
    SDEX <> USDT
      ✓ SELL: Compute pool identiers and prices (738 ms)
      ✓ BUY: Compute pool identiers and prices (638 ms)
    SDEX <> WETH
      ✓ SELL: Compute pool identiers and prices (661 ms)
      ✓ BUY: Compute pool identiers and prices (627 ms)
    SDEX <> ETH
      ✓ SELL: Compute pool identiers and prices (623 ms)
      ✓ BUY: Compute pool identiers and prices (635 ms)
    WETH <> WBTC
      ✓ SELL: Compute pool identiers and prices (629 ms)
      ✓ BUY: Compute pool identiers and prices (656 ms)
  Smardex Integration Tests on arbitrum
    ✓ getTopPoolsForToken (340 ms)
    SDEX <> USDC
      ✓ SELL: Compute pool identiers and prices (707 ms)
      ✓ BUY: Compute pool identiers and prices (460 ms)
    SDEX <> WETH
      ✓ SELL: Compute pool identiers and prices (726 ms)
      ✓ BUY: Compute pool identiers and prices (444 ms)
    SDEX <> ETH
      ✓ SELL: Compute pool identiers and prices (448 ms)
      ✓ BUY: Compute pool identiers and prices (426 ms)
    SDEX <> WBTC
      ✓ SELL: Compute pool identiers and prices (445 ms)
      ✓ BUY: Compute pool identiers and prices (438 ms)
    ARB <> USDC
      ✓ SELL: Compute pool identiers and prices (665 ms)
      ✓ BUY: Compute pool identiers and prices (403 ms)
  Smardex Integration Tests on bsc
    ✓ getTopPoolsForToken (216 ms)
    SDEX <> USDT
      ✓ SELL: Compute pool identiers and prices (128 ms)
      ✓ BUY: Compute pool identiers and prices (124 ms)
    bBTC <> SDEX
      ✓ SELL: Compute pool identiers and prices (120 ms)
      ✓ BUY: Compute pool identiers and prices (125 ms)
    USDT <> WBNB
      ✓ SELL: Compute pool identiers and prices (126 ms)
      ✓ BUY: Compute pool identiers and prices (122 ms)
    ETH <> SDEX
      ✓ SELL: Compute pool identiers and prices (122 ms)
      ✓ BUY: Compute pool identiers and prices (125 ms)
    USDT <> BNB
      ✓ SELL: Compute pool identiers and prices (121 ms)
      ✓ BUY: Compute pool identiers and prices (130 ms)
    RVF <> SDEX
      ✓ SELL: Compute pool identiers and prices (128 ms)
      ✓ BUY: Compute pool identiers and prices (130 ms)
  Smardex Integration Tests on polygon
    ✓ getTopPoolsForToken (336 ms)
    USDC <> SDEX
      ✓ SELL: Compute pool identiers and prices (461 ms)
      ✓ BUY: Compute pool identiers and prices (409 ms)
    SDEX <> WETH
      ✓ SELL: Compute pool identiers and prices (420 ms)
      ✓ BUY: Compute pool identiers and prices (479 ms)
    USDC <> WMATIC
      ✓ SELL: Compute pool identiers and prices (447 ms)
      ✓ BUY: Compute pool identiers and prices (429 ms)
    USDC <> MATIC
      ✓ SELL: Compute pool identiers and prices (449 ms)
      ✓ BUY: Compute pool identiers and prices (424 ms)
    WBTC <> SDEX
      ✓ SELL: Compute pool identiers and prices (434 ms)
      ✓ BUY: Compute pool identiers and prices (408 ms)

Test Suites: 1 passed, 1 total
Tests:       44 passed, 44 total
Snapshots:   0 total
Time:        20.545 s, estimated 23 s```

File: src/dex/smardex/smardex-integration.test.ts [hunk 1]
```diff
@@ -9,12 +9,158 @@ import { Smardex } from './smardex';
...
+        dest: 'BNB',
+        sell: [0, 2_000, 4_000, 6_000],
+        buy: [0, 10, 20, 30],
+      },
+      {
+        src: 'RVF',
+        dest: 'SDEX',
+        sell: [0, 2_000, 4_000, 6_000],
+        buy: [0, 1_000, 2_000],
+      },
```
Comment (asyx21) [inline src/dex/smardex/smardex-integration.test.ts]: Please remove RVF pairs it is a partnership farming campaign

File: tests/constants-e2e.ts [hunk 2]
```diff
@@ -571,6 +575,14 @@ export const Tokens: {
...
       decimals: 6,
     },
+    SDEX: {
+      address: '0xFdc66A08B0d0Dc44c17bbd471B88f49F50CdD20F',
+      decimals: 18,
+    },
+    RVF: {
+      address: '0x872a34Ebb2d54Af86827810EeBC7b9dC6B2144aA',
+      decimals: 18,
+    },
```
Comment (asyx21) [inline tests/constants-e2e.ts]: remove RVF

**Review by asyx21 — COMMENTED** [review]
Great ! You can merge after removal of RVF

File: tests/constants-e2e.ts (hunk 2 above)
Comment (estebgonza) [reply tests/constants-e2e.ts]: Done

**Review by asyx21 — APPROVED** [review]

**Pull request merged**